### PR TITLE
completed frogger-task

### PR DESCRIPTION
**Pull request by sofiichuk**
### Object Oriented JS

[Demo](https://sofiichuk.github.io/frontend-nanodegree-arcade-game) |
[Code base](https://github.com/sofiichuk/frontend-nanodegree-arcade-game/blob/frogger/js/app.js)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by sofiichuk** [comment]
dear bot, dear mentors.
i guess my code meats most of recomendations.
except:
initial bug positions. i could've pretend calculating, but they are what they are - uncalculatable and inavitable;
also i used player = new Player instead of Object.create, cause i couldn't handle it;
and maybe i could separate smaller functions - like reset - from win/lose functions. i tried and i loosed;
if it's ok - it's ok.
and if i have to improve that - i will

File: submissions/sofiichuk/frogger-task/app.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    }
+    this.detectCollision();
+};
+
+Enemy.prototype.detectCollision = function () {
+    if (player.x <= this.x + character.width &&
+        player.x + character.width >= this.x &&
+        player.y <= this.y + character.height &&
+        player.y + character.height >= this.y) {
+        player.reset()
```
Comment (OleksiyRudenko) [inline submissions/sofiichuk/frogger-task/app.js:56]: Item 4.iii of the requirements.

File: submissions/sofiichuk/frogger-task/app.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+};
+
+// ===========================================================
+
+const player = new Player(initialPlayerPosition.x, initialPlayerPosition.y);
+
+// const Enemy = function (x, y, speed)
+const enemy1 = new Enemy(0, 225, 100);
+const enemy2 = new Enemy(0, 142, 200);
+const enemy3 = new Enemy(0, 60, 150);
```
Comment (OleksiyRudenko) [inline submissions/sofiichuk/frogger-task/app.js:133]: Mystery numbers. Item 3 (all subitems) of the requirements.
Also not DRY. Imagine you are tasked to add 10 more enemies. Minimize the effort.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@sofiichuk nice start.
Needs improvements.
Pixel values are definitely calculable based on icons dimensions and playfield row numbers.
This would require certain research.
For example, how did you come up with numbers for `initialPlayerPosition`?
Also how river height is different from cell height?

File: submissions/sofiichuk/frogger-task/app.js (hunk 1 above)
Comment (sofiichuk) [reply submissions/sofiichuk/frogger-task/app.js]: replaced global variable's properties with enemy's own

File: submissions/sofiichuk/frogger-task/app.js (hunk 2 above)
Comment (sofiichuk) [reply submissions/sofiichuk/frogger-task/app.js]: no more mystery numbers.
no mo repetitions.
p.s. also fixed two bugs that had been found during UAT.
p.p.s. please, review .this once again

File: submissions/sofiichuk/frogger-task/app.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+// ===========================================================
+
+const Enemy = function (x, y, speed) {
+    this.x = x;
+    this.y = y;
+    this.width = character.width;
+    this.height = character.height;
+    this.speed = speed;
+    this.rival = player;
```
Comment (OleksiyRudenko) [inline submissions/sofiichuk/frogger-task/app.js:57]: `player` is still a global variable.
What blocks you from treating in the same way as you treat speed or coordinates?

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@sofiichuk global variables seem to be supernatural... while they are not.

File: submissions/sofiichuk/frogger-task/app.js (hunk 3 above)
Comment (sofiichuk) [reply submissions/sofiichuk/frogger-task/app.js]: did that

**Review by OleksiyRudenko — APPROVED** [review]
@sofiichuk great job!

**Pull request merged**